src/plugins/intel_cpu/src/transformations/snippets/x64/op/brgemm_cpu.cpp

```diff
@@ -35,33 +37,48 @@ BrgemmCPU::BrgemmCPU(const Output<Node>& A, const Output<Node>& B, const Output<
     set_input_port_descriptor({0, offset_b}, 1);
     set_output_port_descriptor({0, offset_c}, 0);
     set_input_port_descriptor({0, offset_scratch}, 2);
-    constructor_validate_and_infer_types();
+    custom_constructor_validate_and_infer_types(layout_a, layout_b, layout_c);
+}
+
+void BrgemmCPU::custom_constructor_validate_and_infer_types(std::vector<size_t> layout_a, std::vector<size_t> layout_b, std::vector<size_t> layout_c) {
+    INTERNAL_OP_SCOPE(BrgemmCPU_constructor_validate_and_infer_types);
+    validate_inputs();
+
+    // During ctor call, BrgemmCPU doesn't know his port descriptors.
+    // So we use port descs from source inputs
+    const auto brgemm_copy = is_with_data_repacking() ? get_brgemm_copy() : nullptr;
+    const auto planar_input_shapes =
+        std::vector<ov::PartialShape>{ ngraph::snippets::utils::get_reordered_planar_shape(get_input_partial_shape(0), std::move(layout_a)),
```

Comment:
Minor: Here `std:: move` is redundant because `get_reordered_planar_shape` accepts only const reference, so no actual move will be performed.

Reply:
Right, thanks for the catch!